src/theory/arith/constraint.cpp

```diff
@@ -536,6 +526,16 @@ bool Constraint::hasSimpleFarkasProof() const
   return true;
 }
 
+bool Constraint::isPossiblyTightenedAssumption() const
+{
+  // ... that antecdent must be an assumption ...
+  return (isAssumption()
+          || (hasIntTightenProof()
+              && getConstraintRule().d_antecedentEnd != AntecedentIdSentinel
+              && d_database->getAntecedent(getConstraintRule().d_antecedentEnd)
+                     ->isAssumption()));
```

Comment:
I would suggest to unroll the logic into code...
```suggestion
  
  if (isAssumption()) return true;
  if (!hasIntTightenProof()) return false;
  if (getConstraintRule().d_antecedentEnd == AntecedentIdSentinel) return false;
  return d_database->getAntecedent(getConstraintRule().d_antecedentEnd)
                     ->isAssumption();
```